### PR TITLE
chore: setup changeset

### LINKED ISSUE #27: Introduce Changesets

### PR DESCRIPTION
**Pull request by sor4chi**
close #27 

Hi, @yusukebe!
I configured changeset.

You have to add `NPM_TOKEN` for releasing CI and grant RW access to Github Action.
And please consider installing [changeset-bot](https://github.com/apps/changeset-bot) on this repository.

README referred to [here](https://github.com/honojs/middleware/blob/0232214d4b6e1218ae19a2c5d11783ba2f67f749/README.md)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
```
Comment (yusukebe) [inline .github/workflows/release.yml:16]: What about using `v4`?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Setup Node.js 18.x
+        uses: actions/setup-node@v2
```
Comment (yusukebe) [inline .github/workflows/release.yml:19]: It also can be `v4`.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js 18.x
```
Comment (yusukebe) [inline .github/workflows/release.yml:18]: I think it's better to use `20.x`.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18.x
```
Comment (yusukebe) [inline .github/workflows/release.yml:21]: Same as above. It can be `20.x`.

**Review by yusukebe — COMMENTED** [review]
@sor4chi 

I've left some comments. Please check them!

**Comment by sor4chi** [comment]
@yusukebe 
Thanks! I fixed :-)

**Comment by yusukebe** [comment]
@sor4chi 

You are welcome! I'll merge this now.

**Comment by yusukebe** [comment]
@sor4chi

One thing left to do. Could you create a `build` command in the `package.json` at the top level? You can reference [this](https://github.com/honojs/middleware/blob/main/package.json#L10-L26), although you don't necessarily have to use `npm-run-all`.

**Comment by sor4chi** [comment]
@yusukebe How about this?

File: package.json [hunk 5]
```diff
@@ -10,7 +10,13 @@
   "engines": {
     "node": ">=18.14.1"
   },
+  "scripts": {
+    "build:cloudflare-pages": "yarn workspace @hono/vite-cloudflare-pages build",
+    "build:dev-server": "yarn workspace @hono/vite-dev-server build"
+  },
```
Comment (yusukebe) [inline package.json:16]: We have to make `build` command because it will do `yarn build` in a GitHub release flow.

**Comment by sor4chi** [comment]
@yusukebe Okay, completed

File: package.json (hunk 5 above)
Comment (yusukebe) [reply package.json:17]: @sor4chi 

How about using `yarn workspaces foreach` instead of `npm-run-all`.

https://v3.yarnpkg.com/cli/workspaces/foreach

Comment (sor4chi) [reply package.json:17]: Wow, i didn't know the feature!
You mean we dont nessary to configure build:*?
We only configure `build: yarn workspaces foreach run build`, right?

Comment (yusukebe) [reply package.json:17]: Yes. I think we don't need to build them in parallel, so we can use the `foreach` method of yarn.

Comment (sor4chi) [reply package.json:15]: Okay, I fixed

Comment (sor4chi) [reply package.json:15]: yes, we have to use berry option, right?

Comment (yusukebe) [reply package.json:15]: I think `@yarnpkg/plugin-workspace-tools` is needed.

https://v3.yarnpkg.com/cli/workspaces/foreach

**Review by yusukebe — APPROVED** [review]

**Comment by yusukebe** [comment]
@sor4chi 

Thanks! Let's go with this.

**Comment by yusukebe** [comment]
@sor4chi 

Works well! https://github.com/honojs/vite-plugins/actions/runs/6891680942

**Pull request merged**